added tutorial for qr-code-terminal plugin

shwetalsoni opened the pull request:
Added tutorial for `QR code terminal plugin`. 

![Screenshot from 2021-06-27 13-33-11](https://user-images.githubusercontent.com/57187745/123537345-8ee83100-d74c-11eb-86ea-598b50031186.png)
![Screenshot from 2021-06-27 13-33-40](https://user-images.githubusercontent.com/57187745/123537348-90b1f480-d74c-11eb-874f-921305a0e4f5.png)
![Screenshot from 2021-06-27 13-33-56](https://user-images.githubusercontent.com/57187745/123537350-914a8b00-d74c-11eb-9fd7-ea41eaa9c314.png)
![Screenshot from 2021-06-27 13-34-09](https://user-images.githubusercontent.com/57187745/123537352-91e32180-d74c-11eb-80c7-c2d966cbe005.png)
![Screenshot from 2021-06-27 13-34-23](https://user-images.githubusercontent.com/57187745/123537353-927bb800-d74c-11eb-97a3-156b9ad2a3a6.png)
![Screenshot from 2021-06-27 13-34-35](https://user-images.githubusercontent.com/57187745/123537355-93144e80-d74c-11eb-9427-bb2fcc5acf23.png)

shwetalsoni commented:
Tag #891

proudofsimin commented on docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md:
````diff
@@ -0,0 +1,72 @@
+npm i wechaty-plugin-contrib
+```
+
+## 2. Integrating the plugin
````
remove the v-ing form. Use imperative sentence.

proudofsimin commented on docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md:
````diff
@@ -0,0 +1,72 @@
+bot.start()
+```
+
+## 3. Running the bot
````
remove the v-ing form. Use imperative sentence.

proudofsimin reviewed (changes requested):
Hi, could you explain more about this plugin at the beginning? Is it the users (developers) or customers will scan the QR code? What will the developers receive when someone scan the QR code? And what will it show on phone after scanning the QR code?

shwetalsoni commented:
> Hi, could you explain more about this plugin at the beginning? Is it the users (developers) or customers will scan the QR code? What will the developers receive when someone scan the QR code? And what will it show on phone after scanning the QR code?

Hii simin,
Actually this plugin is just for demonstrating that how a user can use this plugin in their bot. So, now on scanning this qr code it wont do anything.

I think for better explanation I can provide a conclusion at end reffering to ding dong bot because there we have used this plugin. On scanning that qr code users can connect to whatsapp and wechat.

What do you think or is their any better way to do this ?

proudofsimin reviewed (commented):
Hi @shraddhavp,

Let me clear up. So in here, you use the plugin to generate a QR code first, and then you are required to use the events (are they called events?) to define the function for it. Am I right? I agree with your suggestion that mentioning a real example really helps me/user to clear it up. Thank you.

shwetalsoni commented:
Hii @proudofsimin
 I have changed ing(s) to imperative form. 
I have also added conclusion section at end describing about the application of `QR code terminal` plugin. 
I have updated the screenshots for better understanding.
Thank you :)

shwetalsoni commented:
> Hi @shraddhavp,
> 
> Let me clear up. So in here, you use the plugin to generate a QR code first, and then you are required to use the events (are they called events?) to define the function for it. Am I right? I agree with your suggestion that mentioning a real example really helps me/user to clear it up. Thank you.

yeah, you get it right. I have made the changes now. You can review it. 
Btw it's not @shraddhavp :sweat_smile:

proudofsimin commented:
> Hii @proudofsimin
> I have changed ing(s) to imperative form.
> I have also added conclusion section at end describing about the application of `QR code terminal` plugin.
> I have updated the screenshots for better understanding.
> Thank you :)

Thanks for the modification and adding the starter bot. Could you put the "create the starter bot" in step one like #963 ? Thank you.

P.s. sorry for linking the wrong name...

proudofsimin reviewed (approved):
Since it's only minor modification left, approve first.

shwetalsoni commented:
> > Hii @proudofsimin
> > I have changed ing(s) to imperative form.
> > I have also added conclusion section at end describing about the application of `QR code terminal` plugin.
> > I have updated the screenshots for better understanding.
> > Thank you :)
> 
> Thanks for the modification and adding the starter bot. Could you put the "create the starter bot" in step one like #963 ? Thank you.
> 
> P.s. sorry for linking the wrong name...

I have added this step

lijiarui commented on docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md:
```diff
@@ -0,0 +1,108 @@
+
+### 1. Create a starter bot
+
+Follow the instructions on the **Starter Bot** page to create the foundation of a Wechaty bot.
```
I think we should add a starter bot link here for **Starter Bot**

shwetalsoni replied:
Yeah, I will add the link here. Started Bot will be under `examples/basic/`. We have this section in the upcoming weeks in our proposal. So once the starter bot is added and merged we can provide the link here.

For now, I'm thinking to link the `examples/basic` only. Will that be fine?

lijiarui replied:
I think you could file a new issue for this. Once you create a new issue, I will ignore this part.

lijiarui commented on docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md:
```diff
@@ -0,0 +1,108 @@
+
+Now, you are done with the integration of `QR code terminal plugin` with your bot. For running the bot you have to **export/set** an environment variable with the type of puppet to use and then run the bot.
+
+#### Linux/macOS
```
I think we should unify the style here, you did a good job at #888 , it looks like this：

![image](https://user-images.githubusercontent.com/6419514/124647062-0d20a200-dec8-11eb-863c-8cff0c1fd114.png)

You can use code like this:

```markdown
<Tabs
  groupId="operating-systems"
  defaultValue="linux"
  values={[
    { label: 'Linux',   value: 'linux', },
    { label: 'macOS',   value: 'mac', },
    { label: 'Windows', value: 'windows', },
  ]
}>

<TabItem value="linux">

```

lijiarui commented on docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md:
```diff
@@ -0,0 +1,108 @@
+You can use this `QR code terminal` plugin for building the bots where you have to connect to a device. You can refer to [Ding Dong bot](../getting-started/hard-way.mdx). In ding dong bot you have to scan the generated QR code to connect to **Whatsapp/Wechat**.
+
+## References
+
```
Add video link reference here: https://www.youtube.com/watch?v=tfGZXoe_aA4

lijiarui commented on docusaurus/docs/using-plugin-with-wechaty/qr-code-terminal.md:
```diff
@@ -0,0 +1,108 @@
+## References
+
+* [GitHub repository of Wechaty Plugin Contrib](https://github.com/wechaty/wechaty-plugin-contrib)
+
```
Pls remove blank lines here.

lijiarui reviewed (changes requested):
Pls change as I request.

shwetalsoni commented:
@lijiarui I have made the suggested changes except for the starter bot link which I'll add later when the starter bot is created and merged.

lijiarui commented:
Good job! After you file a new issue for [this](https://github.com/wechaty/wechaty.js.org/pull/969#discussion_r665081568), I will approve this PR.

shwetalsoni commented:
@lijiarui I have created issue #1032 for keeping the record of all the features that need to be added later mentioning all the respective PR's.

lijiarui reviewed (approved):
Good job！ Appoved
